File: FE/e-pos-fe/src/services/CategoryService.ts
import api from "@/lib/api";


export class CategoryService {
    static async create(formData: FormData) {
        return await api.post('/categories', formData, {
            headers: {
                'Content-Type': 'multipart/form-data',
            },
        });
    }

    static async getAll(page = 0, size = 10) {
        const response = await api.get(`/categories?page=${page}&size=${size}`);
        return response.data.data;
    }

    static async getByUuid(uuid: string) {
        const response = await api.get(`/categories/${uuid}`);
        return response.data.data;
    }

    static async update(uuid: string, formData: FormData) {
        return await api.put(`/categories/${uuid}`, formData, {
            headers: {
                'Content-Type': 'multipart/form-data',
            },
        });
    }

    static async delete(uuid: string) {
        return await api.delete(`/categories/${uuid}`);
    }


}

File: BACKEND/E-POS/src/main/java/com/zam/dev/pos/backend/services/CategoryService.java
package com.zam.dev.pos.backend.services;

import com.zam.dev.pos.backend.dto.requests.CategoryRequest;
import com.zam.dev.pos.backend.entities.Category;
import com.zam.dev.pos.backend.entities.Tenant;
import org.springframework.data.domain.Page;

public interface CategoryService {

    Category create(CategoryRequest request, Tenant tenant);

    Page<Category> getAll(int page, int size, Tenant tenant);

    Category getByUuid(String uuid, Tenant tenant);

    Category update(String uuid, CategoryRequest request, Tenant tenant);

    void delete(String uuid, Tenant tenant);
}